### DISCUSSION TITLE
verifyClient function or alternative

### DISCUSSION THREAD

**Discussion by subversivo58** [uNetworking/uWebSockets.js#108] [Q&A] [answered]
Hello, thanks for your library

I followed an earlier version which had the option to check the client ... if the verification failed, called the `abortConnection(socket, code, name)` function

```javascript
const http = require('http')
const WebSocketServer = require('uws').Server
const PORT = proccess.env.PORT || 3000

const server = http.createServer((req, res) => {
    // do stuff...
})

const WS = new WebSocketServer({
    server: server,
    verifyClient (info, cb) {
        // for example ever fails
        cb(false, 401, 'Unauthorized')
    }
})

WS.on('connection', function(ws) {
    // never show
    console.log(ws)
    
    ws.on('message', data => {})
    ws.on('error', CloseEventCode => {})
})

server.listen(PORT)
```

The connection was aborted before arriving at the event **onconnection** and in front-end show this in console:

![fail-ws](https://user-images.githubusercontent.com/6475273/54592932-1e4df800-4a0c-11e9-9e7a-7de59c97963b.png)
![fail-ws2](https://user-images.githubusercontent.com/6475273/54592942-2443d900-4a0c-11e9-83ed-f72dcddb0f6e.png)

-----------------------

My question: is it possible to reproduce the same behavior in the current library?

**Comment by ghost** [comment]
No, but you can close the websocket in the open handler if you don't find the http request suitable.

**Comment by ghost** [comment, marked as answer]
UpgradeAsync and UpgradeSync examples shows this feature now
